src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEMENTS = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ELEMENTS];
+        values = (V[]) new Object[MAX_ELEMENTS];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getKeyIndex(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
+        int index = getKeyIndex(key);
+        if (index != -1) {
+            return values[index];
+        }
         return null;
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    private int getKeyIndex(K key) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] != null && keys[i].equals(key) || keys[i] == key) {
```

Comment:
For better readability, you can move the key to a separate variable before checking.
```suggestion
            K currentKey = keys[i];
            if (currentKey != null && currentKey.equals(key) ||currentKey == key) {
```